sql/core/src/main/scala/org/apache/spark/sql/sources/interfaces.scala

```diff
@@ -565,6 +565,7 @@ abstract class HadoopFsRelation private[sql](maybePartitionSpec: Option[Partitio
       filters: Array[Filter],
       inputPaths: Array[String],
       broadcastedConf: Broadcast[SerializableConfiguration]): RDD[Row] = {
+    refresh()
```

Comment:
I agree we'd better provide our own `InputFormat`, at least we minimize the refreshed dir. But it probably requires lots of code change, can we do that in a separated PR?